schema/mysql/schema.sql

```diff
@@ -156,7 +160,12 @@ BEGIN
   END LOOP;
   CLOSE cur;
 
-  SET result = 100 * (total_time - problem_time) / total_time;
+  -- row count "1" because of the faked ending result used for the
+  -- cursor loop, whose result set is never used.
+  IF rowCounts > 1 THEN
```

Comment:
This would also count downtime rows returned by that query which I doubt you want. But also if this would just count the state rows, this makes the assumption that if there's no event in the interval, there's no valid result to report. Problem with this is that if nothing happens on a checkable, there is no history. So for something that's 100% up, had no downtimes, this would now no longer return a value.